zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -839,12 +839,17 @@ private void resetConnectionThreadCount() {
      */
     public class Listener extends ZooKeeperThread {
 
+        private final int portBindMaxRetry;
         volatile ServerSocket ss = null;
 
         public Listener() {
             // During startup of thread, thread name will be overridden to
             // specific election address
             super("ListenerThread");
+
+            // maximum retry count while trying to bind to election port
+            // see ZOOKEEPER-3320 for more details
+            portBindMaxRetry = Integer.getInteger("zookeeper.electionPortBindRetry", 3);
```

Comment:
I'd recommend adding an info log that says what this property is set to when it reads and some validation in place to make sure the property is not negative.

Overall change seems reasonable - let's see what the others think.

Reply:
Add logs and validation